engine.go

```diff
@@ -1,16 +1,139 @@
 package fuego
 
-func NewEngine() *Engine {
-	return &Engine{
-		OpenAPI:      NewOpenAPI(),
-		ErrorHandler: ErrorHandler,
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+	"os"
+	"path/filepath"
+)
+
+func NewEngine(config ...OpenAPIConfig) *Engine {
+	if len(config) > 1 {
+		panic("config should not be more than one")
+	}
+	engine := &Engine{
+		OpenAPI:       NewOpenAPI(),
+		OpenAPIConfig: defaultOpenAPIConfig,
+		ErrorHandler:  ErrorHandler,
+	}
+	if len(config) > 0 {
+		engine.setOpenAPIConfig(config[0])
 	}
+	return engine
 }
 
 // The Engine is the main struct of the framework.
 type Engine struct {
-	OpenAPI      *OpenAPI
-	ErrorHandler func(error) error
+	OpenAPI       *OpenAPI
+	ErrorHandler  func(error) error
+	OpenAPIConfig OpenAPIConfig
 
 	acceptedContentTypes []string
 }
+
+type EngineOpenAPIConfig struct {
+	// If true, the engine will not print messages
+	DisableMessages bool
+	// If true, the engine will not save the OpenAPI JSON spec locally
+	DisableLocalSave bool
+	// Local path to save the OpenAPI JSON spec
+	JsonFilePath string
+	// Pretty prints the OpenAPI spec with proper JSON indentation
+	PrettyFormatJson bool
+}
+
+// OutputOpenAPISpec takes the OpenAPI spec and outputs it to a JSON file
+func (e *Engine) OutputOpenApiSpec() []byte {
+	e.OpenAPI.computeTags()
+
+	// Validate
+	err := e.OpenAPI.Description().Validate(context.Background())
```

Comment:
Either the context is useless, either the function should take it as argument

Reply:
So you're suggesting what exactly? I'm not really a fan of `OutputOpenApiSpec(context.Context)`. We could create ones ourselves in line I guess.

Reply:
Yes,i was suggesting that.

My problem here is that the context created here means the validate code cannot use a real context.

So my point was to either use a context in the openapispec signature, or consider the context in validate is useless

Reply:
It's not critical. And which context OpenAPI Validation should be related to?

Reply:
https://github.com/getkin/kin-openapi/blob/v0.128.0/openapi3%2Fopenapi3.go#L142

I thought the validate thing was in fuego but it is not.

I'll have to dig further

Reply:
https://github.com/search?q=repo%3Agetkin%2Fkin-openapi%20%22context.With%22&type=code

I don't think there is a need for fuego to pass a context, I mean something different than context.Background